From fc503208fc83ca05eb49256002da77f2d71c4056 Mon Sep 17 00:00:00 2001
From: Shawn Guo <shawn.guo@freescale.com>
Date: Sun, 14 Jul 2013 21:35:49 +0800
Subject: [PATCH] ENGR00269945: reset: add dummy device_reset() for
 !CONFIG_RESET_CONTROLLER build

Add dummy device_reset() function for !CONFIG_RESET_CONTROLLER build,
so that we do not have to add #ifdef CONFIG_RESET_CONTROLLER in every
single client device drivers that call the function.

[shawn.guo: cherry-pick commit a9c086f35277 from imx_3.10.y]
Signed-off-by: Shawn Guo <shawn.guo@freescale.com>
---
 include/linux/reset.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/linux/reset.h b/include/linux/reset.h
index 6082247..2a8e982 100644
--- a/include/linux/reset.h
+++ b/include/linux/reset.h
@@ -12,6 +12,13 @@ struct reset_control *reset_control_get(struct device *dev, const char *id);
 void reset_control_put(struct reset_control *rstc);
 struct reset_control *devm_reset_control_get(struct device *dev, const char *id);
 
+#ifdef CONFIG_RESET_CONTROLLER
 int device_reset(struct device *dev);
+#else
+static inline int device_reset(struct device *dev)
+{
+	return 0;
+}
+#endif /* CONFIG_RESET_CONTROLLER */
 
 #endif
-- 
1.8.0

